### PR TITLE
Ranger Modernization & Tweaks

### PR DESCRIPTION
**Pull request by fighterslam**
## About The Pull Request
Updates the Ranger-Class, giving it facilities that feel better without significantly altering the balance. 
image

Major changes: 
Adds a Logistics Specialist to the ship's crew manifest, and enhances the Cargo Bay to accommodate for them. Now they have a desk, a bed, and a locker, all painted in N+S colors! The Cargo Bay now also requires ship access to be opened/closed.
The ship is now a bit more focused on "disaster response" rather than being a generalist, and the tags have been updated to reflect this. I might do some more work on this in the future, to lean into the "ship that can fix both the crew and the ship" aspect. This also leaves the niche of Vigilitas Gunship open for when that's made.
Re-paints the ship to be more in line with Vigilitas colors, and reflect NT Spaceworks' designs (titanium hulls/windows) & updates the prefix to VISV.

Smaller changes:
Adds access locks to the AI core/Comms (engineering access), the Ready Room (requires Ship Access now and a non-intern role), security office, and specialist quarters/rooms
Adds a few internal windows - the Bridge now has a shutter-able window to the atrium, Security gets shutterable windows, and so does Medical. 
Updates the Air Cycler to be more secure and not have any exterior windows, keeping with the Ranger's "no external windows" theme. 
Moves the atmospherics equipment to the engine bays, replacing the equipment room with a proper ready-room for the LP Squad, and a small armory compartment. Also adds a 9mm Saber SMG to the armory. All of the LP hardsuits have been moved into this ready-room, except for the Lieutenant's. 
Enhances Medical by replacing the weird cell with a proper surgical compartment, and uses tinted windows to give the Medical Specialist's bedroom some privacy.
Rearranges the Security Specialist's office by giving them a proper desk, a records console, and a camera monitor, aswell as a much more functional holding cell. 
Makes the AI core & Comms room much prettier (and more secure)
Adjusts the location of the docking port so you no longer need to cross hyperspace
Expands the bridge slightly & adds more shutters to the ship.
Fixes the LP med-spec's hardsuit to be spaceproof again. 
Enhances the LT's living quarters.
Upgrades the Ion Engines to the premium variant - currently, it has the same amount of thrusters as a Kilo, but is significantly heavier. As it'd require significant hull changes to add more thrusters, they've just been upgraded to make it more fitting for a proper response ship. 
Adds an air refilling system, so you don't have to depressurize your air supply in order to refill it. 
Replaces the "Emergency EVA" closets with Fragile Space Suit Storage Units, which are weaker than the previous EVA suits + spawns less supplies. 
Makes it so the LP Lieutenant spawns with the Vigilitas Security Coat rather than the standard NT one, and removes the indie officer stuff from their closet. 
Fixes the Medical Specialist's hardsuit to be spaceproof again. 
Adds GPSes to every specialist's locker, instead of just a few, and gives the AI Core a stationary computer GPS.
Shoves a dresser into Cryo.
Adds External Atmospherics control - you can now control the external atmospherics fittings and also have an ability to actually check the external atmosphere with sensors. Code for it's a bit jank but it works. Are we out of scope yet?
Countless other small changes I've forgotten about as this expands.

## Why It's Good For The Game
Makes the Ranger feel much better to play on - brings the crew together more, makes the areas feel a bit more usable, and helps the Ranger lean into a specialization, rather than being Yet Another NT Generalist.

## Changelog

:cl:
balance: Modernizes and slightly buffs the Ranger. 
/:cl:

### CODE REVIEW

**Comment by Aquidu** [comment]
mods can we merge this guy
papercupjanitor

**Comment by Sadhorizon** [comment]
Hihi! Some stuff. Some copied from discord, some not.
- Ship's very blue, VI's not blue, hallway markings should probably change. Delta goes all-in on orange and it adds to identity I think.
- Prefix, make this a first VISV :3
- It's an NT Spaceworks ship, should probably swap walls for titanium. It's probably meant to be a quality ship, not some civ market garbage.
- Mistake decal in that wall
obraz
- Might be out of scope, but LP Mission Disks did NOT age well. Should probably look over them, remove all the megafauna/syndicate combat stuff.
- These moths/chairs could use layer changes, cuz they are on the same one which tends to be weird.
obraz
- This meant to be NT? Just making sure.
obraz
- Don't think the issue of no starting food has been solved? Would (probably) be nice.
- Why is the docking port so far away?? I know this is also the case on the current Ranger, but it doesn't seem like you changed it.
obraz
- Scrubbers don't seem connected to the outlet.

File: code/game/MapData/shuttles/nanotrasen_ranger.dm [hunk 1]
```diff
@@ -458,5 +341,5 @@
 		/obj/item/disk/holodisk/lp/aid/guard = 1,
 	)
 
-	lootcount = 2
+	lootcount = 4
```
Comment (Sadhorizon) [inline code/game/MapData/shuttles/nanotrasen_ranger.dm:309]: So, how this var works, is that it's the amount of stuff one spawner spawns. I assume it's not intentional?

File: _maps/configs/nanotrasen_ranger.json [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
+	"prefix": "VISV",
 	"map_name": "Ranger-class Aid Vessel",
 	"namelists": [
 		"NANOTRASEN",
 		"NATURAL_AGGRESSIVE",
 		"WEAPONS"
 	],
 	"map_short_name": "Ranger-class",
-	"description": "A Nanotrasen rescue and aid vessel. Equipped with an AI core, moderate combat gear, and equipment fit for rescue and general aid operations. Nanotrasen often deploys these ships in lieu of a proper ERT to aid their allies in the Frontier without committing their full might. The shipowner is the Lieutenant of a Loss Prevention squad, with a Commissioner to aid with operations on the ship proper.",
+	"description": "A Vigilitas Interstellar aid vessel, manufactured by NT Spaceworks. Equipped with a NanoTrasen AI core, moderate combat gear, and equipment fit for rescue and general aid operations. Vigitilas Interstellar often deploys these ships in lieu of a proper ERT to aid their allies in the Frontier without committing their full might. The shipowner is the Lieutenant of a Loss Prevention squad.",
```
Comment (Sadhorizon) [inline _maps/configs/nanotrasen_ranger.json]:
```suggestion
	"description": "A Vigilitas Interstellar aid vessel, manufactured by NT Spaceworks. Equipped with a Nanotrasen AI core, moderate combat gear, and equipment fit for rescue and general aid operations. Vigitilas Interstellar often deploys these ships in lieu of a proper ERT to aid their allies in the Frontier without committing their full might. The shipowner is the Lieutenant of a Loss Prevention squad.",
```

**Comment by Apogee-dev** [comment]
Can you replace the Lieutenant's duster with a Vigilitas HoS coat? As the guy in charge, it fits, and it's better to have him in Vigilitas gear than an indie duster.

**Comment by Sadhorizon** [comment]
There is not a single dresser on the entire ship. Those are good to map in generally.

File: _maps/configs/nanotrasen_ranger.json [hunk 3]
```diff
@@ -1,24 +1,23 @@
 {
 	"$schema": "https://raw.githubusercontent.com/shiptest-ss13/Shiptest/master/_maps/ship_config_schema.json",
+	"prefix": "VISV",
 	"map_name": "Ranger-class Aid Vessel",
 	"faction": "/datum/faction/nt",
```
Comment (Sadhorizon) [inline _maps/configs/nanotrasen_ranger.json]: `"faction": "/datum/faction/nt/vigilitas",`

Comment (FalloutFalcon) [reply _maps/configs/nanotrasen_ranger.json]: I should probably have this tested for..

**Review by Zevotech — CHANGES_REQUESTED** [review]
To start, this is a ship that's been through a lot of iteration and testing. It functions well, and fits well enough as a remap for the Ranger. Most of this review will boil down to fixing errors and design issues.

## Minor Fixes
The Saber SMG in the armory is not actually an NT weapon on a technicality. It is technically Inteq due to being pathed under their weapon, and thus comes with an Inteq colored stock. How you fix this is mostly up to you, be it removing the stock or just changing the weapon.
image

There is a wall intercom in the suit storage holding the RIG suit. I'd reccomend moving it to the wall over the table so this doesn't happen. Also, the RIG suit itself has a spelling error in its description. It should be "A fully heat-resistant suit based on an early RIG hardsuit prototype."
image
image

The "Aid Mission Recruitment Drive" disk has the objective of "The LP will need to convince 4 independent vessels to switch to NTSV flight tags". Due to our rules on false-flagging and ship prefixes, this may need to be changed. If you already talked with someone about this and got a pass for it, let me know and we can discuss it.

Similarly, "Combat Mission Blood Red" may need some changes to discourage players from blindly attempting to rob a syndicate ship. Very few syndicate ships even have a Blood-Red, so you may want to change this to acquiring a Rusted-Red which can be found in ruins or from Ramzi pirates. 

## Design Gripes
If you have a case for anything I bring up here, feel free to discuss it with me.

Filing cabinet on the table with the medkits looks strange. I would remove it or replace it with a wall locker for the medical kits. I don't see what the medical specialist would use this for, given that they have no paper and pen to use within reach of it.
image

Air supply pump should start active so that the airlocks can be used. I've seen a few people forget to turn their distro supply pump on and wonder why their airlocks aren't working.
image

Firelocks underneath an external airlock can interfere with them being used, as they will always close when the airlock opens to space. The firelocks in the hallway are more than enough to stop decompression if it breaches.
image

I'd suggest adding a few fire extinguisher cabinets in the engine nacelles or in the hallways near them. The full-sized lockers are good, but it feels strange for the only place on the ship to have a fire extinguisher cabinet be the operating room.

The decal and the holopad are off-center from eachother...
image

Bridge stairs have decals on them. Looks a bit strange.
image

Lastly and most importantly, What the Hell is up with the sheer volume of handrails on this ship? The ones on the walls are fine, but the ones on the railings and even a table are very odd looking. Just one or two per area on the walls is more than enough.
image
image
image

**Comment by fighterslam** [comment]
All requested changes have been made

**Review by Zevotech — APPROVED** [review]
Vigilitas W

**Comment by fighterslam** [comment]
@Zevotech I think this should be ready for merge or at the very least to be testmerged again. I would like to get this PR out of limbo 🥺

**Review by meemofcourse — APPROVED** [review]
sure

**Pull request merged**